src/pages/index.js

```diff
@@ -32,79 +39,167 @@ const profileDescriptionInput = document.querySelector(
+/* -------------------------------------------------------------------------- */
+/*                            Profile Picture Edit                            */
+/* -------------------------------------------------------------------------- */
 
-  return card.getView();
+function loadInitialProfilePicture() {
+  api
+    .loadInitialProfileData()
+    .then((profileData) => {
+      profilePictureImage.src = profileData.avatar;
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+}
+
+loadInitialProfilePicture();
+
+/* -------------------------------------------------------------------------- */
+/*                               Event Hanlders                               */
+/* -------------------------------------------------------------------------- */
+
+function handleEditProfileSubmit(formData) {
+  modalEditForm.setSubmitText(true);
+  api
+    .editProfileData(formData)
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      modalEditForm.close();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => modalEditForm.setSubmitText(false));
 }
 
-function handleEditProfileSubmit(data) {
-  userInfo.setUserInfo(data);
-  modalEditForm.close();
+function handleEditProfilePictureSubmit(data) {
+  profilePictureImage.src = data.url;
+  editProfilePicture.setSubmitText(true);
+  api
+    .updateProfilePicture(data.url)
+    .then(() => {
+      editProfilePicture.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 
 function handleImageClick(data) {
   modalPreviewImage.open(data);
 }
 
+function handleRemoveCard(data) {
+  removeCardModal.open();
+  removeCardModal.setSubmithandler(() => {
+    removeCardModal.setSubmitText(true);
+    api
+      .deleteNewCard(data._id)
+      .then(() => {
+        removeCardModal.close();
+        data._handleDeleteButton();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => removeCardModal.setSubmitText(false));
+  });
+}
+
 function handleAddCardSubmit(data) {
+  addCardImageModal.setSubmitText(true);
   const name = data.title;
   const link = data.url;
+
   if (name && link) {
     const newCard = {
       name: name,
       link: link,
     };
 
-    const newCardEl = renderCard(newCard);
-
-    section.addItem(newCardEl);
+    api
+      .addNewCard(newCard)
+      .then((newCardData) => {
+        renderCard(newCardData);
+      })
+      .catch((error) => {
+        console.error(error);
+      })
+      .finally(() => addCardImageModal.setSubmitText(false));
   }
 }
 
+function handleCardLike(data) {
+  const isLiked = data.isLiked();
+  const cardId = data.getId();
+
+  api
+    .toggleCardLike(cardId, isLiked)
+    .then((updatedData) => {
+      data.setLikes(updatedData.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors